src/django_components/dependencies.py

```diff
@@ -0,0 +1,764 @@
+"""All code related to management of component dependencies (JS and CSS scripts)"""
```

Comment:
As the comment says, this file contains all the code related to managing the JS / CSS dependencies.

I tried to organize it into sections what flow in chronological order, as there's a lot to it:
1. Caching of the component's inlined JS / CSS
2. Inserting of the `<!-- _RENDERED -->` comment into the HTML, so we know which components were used to render an HTML, so we know which JS and CSS we need to load.
3. Extracting of the `<!-- _RENDERED -->` comments from HTML using `render_dependencies` - This is what was previously done in the middleware, but ofc it was expanded upon here.
4. Defining a Django view that we can request to get component's inlined JS / CSS. That way we can treat the inlined JS / CSS similarly to the JS / CSS defined in `Media.js/css`
5. Defining the middleware that wraps `render_dependencies` - This is the same middleware as was previously defined in `middleware.py`